### PR TITLE
Feat/folder architecture

### PR DESCRIPTION
**Pull request by SushiRoll53**

### CODE REVIEW

File: lib/tags/custom_tag/custom_tag.dart [hunk 1]
```diff
@@ -73,17 +75,22 @@ class CustomTag extends PBTag implements PBInjectedIntermediate {
...
   /// Variable that dictates in what directory the tag will be generated.
-  static const DIRECTORY_GEN = 'controller/tag';
+  static String DIRECTORY_GEN = GetIt.I.get<PathService>().widgetsRelativePath;
+  static String DIRECTORY_CUSTOM =
+      GetIt.I.get<PathService>().customRelativePath;
 
   @override
   String generate(PBIntermediateNode source, PBContext context) {
+    var customDirectory =
+        DIRECTORY_GEN + '/' + context.tree.name + '/' + DIRECTORY_CUSTOM;
```
Comment (ivan-015) [inline lib/tags/custom_tag/custom_tag.dart]: For this part, it may be better to use the path.join() method to have more reliable paths

Comment (SushiRoll53) [reply lib/tags/custom_tag/custom_tag.dart]: You are correct, let me do this change

File: lib/generation/generators/value_objects/file_structure_strategy/pb_file_structure_strategy.dart [hunk 2]
```diff
@@ -28,10 +30,12 @@ abstract class FileStructureStrategy implements CommandInvoker {
   ///
   ///The views is anything that is not a screen, for example, symbol masters
   ///are going to be generated in this folder if not specified otherwise.
-  static final RELATIVE_VIEW_PATH = 'lib/widgets/';
+  static final RELATIVE_VIEW_PATH =
```
Comment (ivan-015) [inline lib/generation/generators/value_objects/file_structure_strategy/pb_file_structure_strategy.dart]: I believe `RELATIVE_VIEW_PATH` and `RELATIVE_SCREEN_PATH` are flipped in terms of their value. 

`RELATIVE_VIEW_PATH` should probably be renamed to `RELATIVE_WIDGET_PATH`. The wrong name was probably given to the variable when it was first created

**Review by ivan-015 — APPROVED** [review]
lGTM

**Pull request merged**